### PR TITLE
Sound effect refactor

### PR DESCRIPTION
**Pull request by mj0ln1r42**

### CODE REVIEW

File: include/games/rocket/rocket_game.h [hunk 1]
```diff
@@ -298,13 +281,15 @@ class RocketGame : public Game {
...
-          rocket.SetBoost(rocket.physics.Thrust + 5);  // just keep boosting up
-          rocket.Move(false);  // let it boost off the screen
+        if (rocket.physics.location.y < display->size.y * 2) {
+          rocket.SetBoost(rocket.physics.thrust.y +
+                          5);  // just keep boosting up
+          rocket.physics.respect_edges = false;
+          rocket.Move();  // let it boost off the screen
 
-          // shift stars and target down according to Rocket Thrust up to 10
+          // shift stars and target down according to Rocket thrust up to 10
```
Comment (JonBedell) [inline include/games/rocket/rocket_game.h:290]: lol VSCode updating comments

File: include/engines/physics_info.h [hunk 2]
```diff
@@ -35,87 +32,77 @@ class PhysicsInfo {
...
-    if (Thrust > ThrustMax) {
-      Thrust = ThrustMax;
+  void Move() {
+    if (thrust.y > ThrustMax) {
+      thrust.y = ThrustMax;
     }
 
-    float timeDiff = (float)(millis() - Time) / 1000;
+    const auto now = millis();
+    const float timeDiff = (float)(now - last_move_time) / 1000.0f;
```
Comment (JonBedell) [inline include/engines/physics_info.h:67]: what does the f do here?

File: include/math/vector2d.h [hunk 3]
```diff
@@ -0,0 +1,126 @@
...
+  template <typename U>
+  Vector2D(const Vector2D<U>& other) : x{other.x}, y{other.y} {}
+
+  Vector2D& operator=(const Vector2D& other) {
+    x = other.x;
+    y = other.y;
+    return *this;
+  }
+  template <typename U>
+  Vector2D& operator=(const Vector2D<U>& other) {
```
Comment (JonBedell) [inline include/math/vector2d.h:28]: operator?

**Review by JonBedell — APPROVED** [review]

File: include/engines/physics_info.h (hunk 2 above)
Comment (mj0ln1r42) [reply include/engines/physics_info.h:67]: The f is a way that we can show the compiler that this literal value is meant to be interpreted as a float. I think the #.0 indicates that too but sometimes a compiler can optimize that away so I'm trying to be explicit that it's a float

File: include/math/vector2d.h (hunk 3 above)
Comment (mj0ln1r42) [reply include/math/vector2d.h:28]: Yeah, c++ supports operator overloading! These methods are the behavior that is invoked when we do assignment with the `=` operator or you can overload things like `+=`, `*`, or even `[..]`

File: include/games/rocket/rocket_game.h (hunk 1 above)
Comment (mj0ln1r42) [reply include/games/rocket/rocket_game.h:290]: Oof tell me about it

**Pull request merged**